tractor/_actor.py

```diff
@@ -259,63 +277,91 @@ def get_waitq(self, actorid, cid):
         # TODO: once https://github.com/python-trio/trio/issues/467 gets
         # worked out we'll likely want to use that!
         log.debug(f"Entering msg loop for {chan} from {chan.uid}")
-        async with trio.open_nursery() as nursery:
-            try:
-                async for msg in chan.aiter_recv():
-                    if msg is None:  # terminate sentinel
-                        log.debug(
-                            f"Cancelling all tasks for {chan} from {chan.uid}")
-                        nursery.cancel_scope.cancel()
-                        log.debug(
-                                f"Msg loop signalled to terminate for"
-                                f" {chan} from {chan.uid}")
-                        break
-                    log.debug(f"Received msg {msg} from {chan.uid}")
-                    cid = msg.get('cid')
-                    if cid:  # deliver response to local caller/waiter
-                        await self._push_result(chan.uid, cid, msg)
-                        log.debug(
-                            f"Waiting on next msg for {chan} from {chan.uid}")
-                        continue
-                    else:
-                        ns, funcname, kwargs, actorid, cid = msg['cmd']
+        try:
+            async for msg in chan.aiter_recv():
+                if msg is None:  # terminate sentinel
+                    log.debug(
+                        f"Cancelling all tasks for {chan} from {chan.uid}")
+                    scopes = self._rpc_tasks.pop(chan, None)
+                    if scopes:
```

Comment:
If you do `scopes = self._rpc_tasks.pop(chan, ())` instead, you can skip the `if`.

Reply:
Good point.